### DISCUSSION TITLE
Custom / Html Type for Forms

### DISCUSSION THREAD

**Discussion by musaffarpatel** [PrestaShop/PrestaShop#35130] [Q&A] [answered]
When using symfony forms, is it possible to render some custom html instead of a predefined form type?  For example:

        $builder
            ->add('mycontrol',CustomHtml::class, [
                'label' => $this->trans('Enabled Categories', 'Module.Test.Admin'),
                'content' => '<div>my html</div>',
            ]);

**Comment by kpodemski** [comment, marked as answer]
You could do it cleaner by utilizing `src/PrestaShopBundle/Form/Admin/Type/CustomContentType.php`.

Example of use is in `src/PrestaShopBundle/Form/Admin/Improve/Design/Pages/CmsPageType.php` for `seo_preview` field.

**Comment by musaffarpatel** [comment]
@kpodemski The reference you provided to the CmsPageType was very helpful, much appreciated.
